cmd/clusterctl/internal/test/fake_objects.go

```diff
@@ -43,16 +43,17 @@ import (
 )
 
 type FakeCluster struct {
-	namespace             string
-	name                  string
-	controlPlane          *FakeControlPlane
-	machinePools          []*FakeMachinePool
-	machineDeployments    []*FakeMachineDeployment
-	machineSets           []*FakeMachineSet
-	machines              []*FakeMachine
-	withCloudConfigSecret bool
-	withCredentialSecret  bool
-	topologyClass         *string
+	namespace                                   string
+	name                                        string
+	controlPlane                                *FakeControlPlane
+	machinePools                                []*FakeMachinePool
+	machineDeployments                          []*FakeMachineDeployment
+	machineDeploymentsWithoutBootstrapConfigRef []*FakeMachineDeployment
```

Comment:
Having two sets of machine deployments seems counter-intuitive, can we solve the same use case by adding an attribute at MD deployment level, so we can Do test.NewFakeMachineDeployment("md02").WithStaticBootstrapConfig.

Then also fake MS and fake Machines can detect the setting in the parent object and generate resources accordingly